### DISCUSSION TITLE
Apply beatmap offset to global audio offset

### DISCUSSION THREAD

**Discussion by ArijanJ** [ppy/osu#22537] [Ideas]
Adding a button similar to 'Calibrate using last play' that adds your current beatmap offset to your global offset setting would be nice for people who don't understand or aren't sure which way to move the global offset slider, and set their offset by feel.

**Comment by peppy** [comment]
I tend to agree that this would be a very good option to have. it would allow users to adjust the universal offset from a very convenient location without necessarily having to understand what it is.

If we do this, we'd probably want to improve the display of offsets on that screen to include showing the global one in some form.

**Reply by monochrome22** [reply]
Maybe a swap button.
![swap](https://user-images.githubusercontent.com/122969285/217198473-0c48cb2a-c8ae-4af2-9f19-664fe4eb9025.png)
It should automatically switch back to beatmap instead of global when restarting.

**Reply by peppy** [reply]
That's just going to be completely confusing LOL. They should not switch but add to each other.

**Reply by monochrome22** [reply]
I'm not sure how it would be confusing. 
But the reasoning was that if you want to display global offset in the gameplay intro, you would have to write the text "Global offset", and it would be more convenient if there was a slider for adjusting it so players wouldn't have to pause and go to settings to change it, then there would also have to be a "Calibrate using last play" button, which is the entire point of this discussion. So in the end you would just be creating 2 copies of nearly the same thing, which would clutter the UI, so being able to switch between the two would make sense and be clean. I think we're on different pages on how it's supposed to function and what counts as a clean solution. 

Edit: I misread the original post. Even though I misunderstood, a swap button might be worth considering. Displaying the global offset in gameplay intro might not be good, since it's not specific to beatmap and for most players it would just be at 0. With a swap button it would be out of sight and you could easily adjust global offset without having to go to settings. It could have both buttons, "Calibrate using last play" and "Apply beatmap offset to global offset", so they wouldn't need to be added to settings. All without cluttering the UI.
